operator/pkg/client/roles/client.go

```diff
@@ -64,45 +64,42 @@ func (c *Client) Has(ctx context.Context, role *redpandav1alpha2.RedpandaRole) (
 }
 
 // Create creates a role in the Redpanda cluster.
-func (c *Client) Create(ctx context.Context, role *redpandav1alpha2.RedpandaRole) error {
-	if role == nil || role.Name == "" {
-		return fmt.Errorf("role is nil or has empty name")
-	}
-
+// Returns the list of principals that were successfully synced.
+func (c *Client) Create(ctx context.Context, role *redpandav1alpha2.RedpandaRole) ([]string, error) {
 	// Check if role already exists
 	exists, err := c.Has(ctx, role)
 	if err != nil {
-		return fmt.Errorf("checking if role %s already exists: %w", role.Name, err)
+		return nil, fmt.Errorf("checking if role %s already exists: %w", role.Name, err)
```

Comment:
NIT: In this repository we try to use `github.com/cockroachdb/errors`. I'm not sure why golangci lint doesn't fail as we have rule for that :) https://github.com/redpanda-data/redpanda-operator/blob/03dd39424afd706da4d7b74d07e296a536aeb0d0/.golangci.yml#L38-L43

Never the less if it's not a big hustle for you could you submit new PR that changes any `fmt.Errorf` to `errors.Wrap`? Thank you 🙇‍♂️